pkg/cluster/cluster.go

```diff
@@ -235,9 +235,13 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 				// TODO: we can't handle another upgrade while an upgrade is in progress
 				c.logger.Infof("spec update: from: %v to: %v", c.cluster.Spec, event.cluster.Spec)
 				c.cluster = event.cluster
+
 			case eventDeleteCluster:
+				c.status.SetReason("cluster is deleted by the user")
```

Comment:
There is another logging message
```
"deleting the failed cluster"
```

We'd better have one message for either "failed deleting" and "normal deleting". But definitely not two together.